### PR TITLE
update-codegen.sh: make generators selectable

### PR DESCRIPTION
**Pull request by rinckm**
### Description

The code generation script `hack/update-codegen.sh` got two new options `--gen-clients` and `--gen-mocks` to select what should be generated. If none of the `--gen-*` options is specified, _all_ generators are enabled.

### Submitter checklist

- [ ] Change has been tested (on a back-end cluster)
- [ ] (If applicable) Jira backlog item ID added to the PR title and the [changelog.yaml] entry
- [ ] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [ ] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- [ ] Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- [ ] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [ ] Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There is at least 1 approval for the pull request and no outstanding requests for change
- [ ] All voter checks have passed
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] The Pull Request title is understandable and reflects the changes well
- [ ] The Pull Request description is understandable and well documented
- [ ] [changelog.yaml] entry for this Pull Request has been added
    - [ ] Changelog entry contains all required information
    - [ ] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

**Review by anbrsap — CHANGES_REQUESTED** [review]
The script duplicates many parts of `hack/update-codegen.sh`, which violates the DRY principle. Also refactor `hack/update-codegen.sh` to avoid code duplication.

File: changelog.yaml [hunk 1]
```diff
@@ -51,6 +51,13 @@
 - version: NEXT
   date: TBD
   changes:
+  - type: internal
+    impact: patch
+    title: "Add update-mocks script"
+    description: |-
+      Add a script to regenerate the mocks without regeneration all clients.
```
Comment (rinckm) [inline changelog.yaml:58]:
```suggestion
  - type: internal
    impact: patch
    title: "Add only-mocks mode to the codegen script"
    description: |-
      Add a mode to only regenerate the mocks without regeneration all clients.
```

File: hack/update-codegen.sh [hunk 2]
```diff
@@ -24,6 +25,9 @@ function read_args() {
                 print_usage
                 exit 0
                 ;;
+	    "-m" | "--only-mocks" )
+	        ONLY_MOCKS=1
+		;;
```
Comment (anbrsap) [inline hack/update-codegen.sh:30]: Indentation is inconsistent.

File: hack/update-codegen.sh [hunk 3]
```diff
@@ -51,12 +55,19 @@ function print_usage() {
     echo "      Verifies that the generated code is up-to-date. The existing generated"
     echo "      code will not be touched."
     echo ""
+    echo "  -m, --only-mocks"
+    echo "      Only regenerate mocks"
+    echo ""
```
Comment (anbrsap) [inline hack/update-codegen.sh:60]:
```suggestion
    echo "  -m, --only-mocks"
    echo "      Only regenerate mocks."
    echo ""
```

File: hack/update-codegen.sh [hunk 4]
```diff
@@ -51,12 +55,19 @@ function print_usage() {
     echo "      Verifies that the generated code is up-to-date. The existing generated"
     echo "      code will not be touched."
     echo ""
+    echo "  -m, --only-mocks"
```
Comment (anbrsap) [inline hack/update-codegen.sh:58]: The "only XXX" approach makes it impossible to select more than one type.

I propose to call this option `--mocks`. In the future we could add further options to select other types, e.g. `--tekton-client`. You can combine those options to generate more than one type but not all. If none of those options is specified, all types get generated.

**Review by anbrsap — APPROVED** [review]

**Comment by anbrsap** [comment]
**Squash before merge!**

File: changelog.yaml [hunk 5]
```diff
@@ -51,6 +51,15 @@
 - version: NEXT
   date: TBD
   changes:
+
+  - type: internal
+    impact: patch
+    title: [Developer] update-codegen.sh: make generators selectable
```
Comment (rinckm) [inline changelog.yaml:57]:
```suggestion
    title: "[Developer] update-codegen.sh: make generators selectable"
```

**Pull request merged**